README-deploy-benchmark.md

```diff
@@ -0,0 +1,221 @@
+    - [Data collection](#data-collection)
+- [Teardown](#teardown)
+
+## Deployment
+
+### Prerequisites
+
+- Kubernetes installation: Use [kubespray](https://github.com/opea-project/docs/blob/main/guide/installation/k8s_install/k8s_install_kubespray.md) or other official Kubernetes installation guides.
+- Helm installation: Follow the [Helm documentation](https://helm.sh/docs/intro/install/#helm) to install Helm.
+- Setup Hugging Face Token
+
+  To access models and APIs from Hugging Face, set your token as environment variable.
+
+  ```bash
+  export HF_TOKEN="insert-your-huggingface-token-here"
+  ```
+
+- Prepare Shared Models (Optional but Strongly Recommended)
+
+  Downloading models simultaneously to multiple nodes in your cluster can overload resources such as network bandwidth, memory and storage. To prevent resource exhaustion, it's recommended to preload the models in advance.
+
+  ```bash
+  pip install -U "huggingface_hub[cli]"
+  sudo mkdir -p /mnt/models
+  sudo chmod 777 /mnt/models
+  huggingface-cli download --cache-dir /mnt/models Intel/neural-chat-7b-v3-3
+  export MODEL_DIR=/mnt/models
+  ```
+
+  Once the models are downloaded, you can consider the following methods for sharing them across nodes:
+
+  - Persistent Volume Claim (PVC): This is the recommended approach for production setups. For more details on using PVC, refer to [PVC](https://github.com/opea-project/GenAIInfra/blob/main/helm-charts/README.md#using-persistent-volume).
+  - Local Host Path: For simpler testing, ensure that each node involved in the deployment follows the steps above to locally prepare the models. After preparing the models, use `--set global.modelUseHostPath=${MODELDIR}` in the deployment command.
+
+- Label Nodes
+  ```base
+  python deploy.py --add-label --num-nodes 2
+  ```
+
+### Deployment Scenarios
+
+This deployment supports scalable operations for the following AI examples. You can specify the example name using the `--chart-name` flag:
+
+- **chatqna**
+- **codegen**
+- **codetrans**
+- **faqgen**
+- **docsum**
+
+Note that only **chatqna** has `--with-rerank` support and can run with `--tuned` mode currently.
+
+The example below are based on a two-node setup. You can adjust the number of nodes by using the `--num-nodes` option.
+
+By default, these commands use the `default` namespace. To specify a different namespace, use the `--namespace` flag with deploy, uninstall, and kubernetes command. Additionally, update the `namespace` field in `benchmark.yaml` before running the benchmark test.
+
+For additional configuration options, run `python deploy.py --help`
+
+#### Case 1: Baseline Deployment for Chatqna with Rerank
+
+Deploy Command (with node number, Hugging Face token, model directory specified):
+
+```bash
+python deploy.py --chart-name chatqna --hf-token $HF_TOKEN --model-dir $MODEL_DIR --num-nodes 2 --with-rerank
+```
+
+Uninstall Command:
+
+```bash
+python deploy.py --chart-name chatqna --uninstall
```

Comment:
```suggestion
python deploy.py --chart-name <your-chart-name> --uninstall
```

more general

Reply:
Please look at the new REAMDE, add note to explain that chatqna just an example, give chatqna as an example is to show how to find the yaml file.